dbt_copilot_helper/commands/codebase.py

```diff
@@ -109,14 +111,106 @@ def build(app, codebase, commit):
     ):
-        boto3.client("codebuild").start_build(
+        response = boto3.client("codebuild").start_build(
             projectName=f"codebuild-{app}-{codebase}",
             artifactsOverride={"type": "NO_ARTIFACTS"},
             sourceVersion=commit,
         )
+        build_url = get_build_url_from_arn(response["build"]["arn"])
 
         return click.echo(
-            "Your build has been triggered. Check your build progress in the AWS Console."
+            "Your build has been triggered. Check your build progress in the AWS Console: "
+            f"{build_url}",
         )
 
     return click.echo("Your build was not triggered.")
+
+
+@codebase.command()
+@click.option("--app", help="AWS application name", required=True)
+@click.option("--env", help="AWS Copilot environment", required=True)
+@click.option("--codebase", help="GitHub codebase name", required=True)
+@click.option("--commit", help="GitHub commit hash", required=True)
+def deploy(app, env, codebase, commit):
+    """Trigger a CodePipeline pipeline based deployment."""
+    try:
+        application = load_application(app)
+    except ApplicationNotFoundError:
+        click.secho(
+            f"""The account "{os.environ.get("AWS_PROFILE")}" does not contain the application "{app}"; ensure you have set the environment variable "AWS_PROFILE" correctly.""",
+            fg="red",
+        )
+        raise click.Abort
+
+    if not application.environments.get(env):
+        click.secho(
+            f"""The environment "{env}" either does not exist or has not been deployed.""",
+            fg="red",
+        )
+        raise click.Abort
+
+    ssm_client = boto3.client("ssm")
+    try:
+        codebase_configuration = json.loads(
+            ssm_client.get_parameter(
+                Name=f"/copilot/applications/{application.name}/codebases/{codebase}",
+            )["Parameter"]["Value"]
+        )
+    except (KeyError, ValueError, json.JSONDecodeError, ssm_client.exceptions.ParameterNotFound):
+        click.secho(
+            f"""The codebase "{codebase}" either does not exist or has not been deployed.""",
+            fg="red",
+        )
+        raise click.Abort
+
+    ecr_client = boto3.client("ecr")
+    try:
+        ecr_client.describe_images(
+            repositoryName=f"{application.name}/{codebase_configuration['name']}",
+            imageIds=[{"imageTag": f"commit-{commit}"}],
+        )
+    except ecr_client.exceptions.RepositoryNotFoundException:
+        click.secho(
+            f'The ECR Repository for codebase "{codebase}" does not exist.',
+            fg="red",
+        )
+        raise click.Abort
+    except ecr_client.exceptions.ImageNotFoundException:
+        click.secho(
+            f'The commit hash "{commit}" has not been built into an image, try the '
+            "`copilot-helper codebase build` command first.",
+            fg="red",
+        )
+        raise click.Abort
```

Comment:
It would be nice to extract all the validation out into one or more validate functions.